src/kernel/boot.c

```diff
@@ -810,26 +810,21 @@ BOOT_CODE void bi_finalise(void)
     };
 }
 
-BOOT_CODE static inline pptr_t ceiling_kernel_window(pptr_t p)
+BOOT_CODE static word_t check_available_memory(word_t n_available,
+                                               const p_region_t *available)
 {
-    /* Adjust address if it exceeds the kernel window
-     * Note that we compare physical address in case of overflow.
-     */
-    if (pptr_to_paddr((void *)p) > PADDR_TOP) {
-        p = PPTR_TOP;
+    /* Ensure avail_reg is properly initialized. */
+    for (word_t i = 0; i < ARRAY_SIZE(avail_reg); i++) {
+        avail_reg[i] = REG_EMPTY;
     }
-    return p;
-}
 
-BOOT_CODE static bool_t check_available_memory(word_t n_available,
-                                               const p_region_t *available)
-{
     /* The system configuration is broken if no region is available. */
     if (0 == n_available) {
         printf("ERROR: no memory regions available\n");
-        return false;
+        return 0;
     }
 
+    word_t cnt = 0;
     printf("available phys memory regions: %"SEL4_PRIu_word"\n", n_available);
```

Comment:
Not from this PR, but if we're changing the logging, we might keep the capitalisation consistent with the `Bootstrapping kernel` etc messages, which start with upper case.

```suggestion
    printf("Available phys memory regions: %"SEL4_PRIu_word"\n", n_available);
```